pymeasure/instruments/hcp/tc038d.py

```diff
@@ -55,89 +53,83 @@ class TC038D(Instrument):
-        if got[1] == self.functions['read']:
-            length = got[2]
-            # data length, 2 Byte CRC
-            read = self.read_bytes(length + 2)
-            if read[-2:] != bytes(CRC16(got + read[:-2])):
-                raise ConnectionError("Response CRC does not match.")
-            return read[:-2]
-        else:  # an error occurred
-            end = self.read_bytes(2)  # empty the buffer
-            if got[2] == 0x02:
-                raise ValueError(f"The read start address {address} is invalid.")
-            if got[2] == 0x03:
-                raise ValueError(f"The number of elements {count} exceeds the allowed range.")
-            raise ConnectionError(f"Unknown read error. Received: {got} {end}")
-
-    def writeMultiple(self, address, values):
-        """Write multiple variables."""
-        data = [self.address]
-        data.append(self.functions['writeMultiple'])  # function code
-        data += [address >> 8, address & 0xFF]  # 2B address
-        if isinstance(values, int):
-            data += [0x0, self.byteMode // 2]  # 2B number of elements
-            data.append(self.byteMode)  # 1B number of write data
-            for i in range(self.byteMode - 1, -1, -1):
-                data.append(values >> i * 8 & 0xFF)
-        elif isinstance(values, collections.abc.Sequence):
+    def write(self, command):
+        """Write a command to the device.
+
+        :param str command: comma separated string of:
+            - the function read (R) or write (W),
+            - the hexadecimal address to write to (e.g. 0x106),
+            - the values (comma separated) to write
+            - or the number of elements to read (defaults to 1).
+        """
+        function, address, *values = command.split(",")
+        data = [self.address]  # 1B device address
+        data.append(self.functions[function])  # 1B function code
+        address = int(address, 16)  # register address
+        data += [address >> 8, address & 0xFF]  # 2B register address
+        if data[1] == self.functions['writeMultiple']:
             elements = len(values) * self.byteMode // 2
             data += [elements >> 8, elements & 0xFF]  # 2B number of elements
-            data.append(len(values) * self.byteMode)  # 1B number of write data
+            data.append(elements * 2)  # 1B number of bytes to write
             for element in values:
-                for i in range(self.byteMode - 1, -1, -1):
-                    data.append(element >> i * 8 & 0xFF)
-        else:
-            raise TypeError(("Values has to be an integer or an iterable of "
-                             f"integers. values: {values}"))
+                data.extend(int(element).to_bytes(self.byteMode, byteorder="big", signed=True))
+        elif data[1] == self.functions['read']:
+            count = int(values[0]) * self.byteMode // 2 if values else self.byteMode // 2
+            data += [count >> 8, count & 0xFF]  # 2B number of elements to read
```

Comment:
```suggestion
            data.extend(count.to_bytes(2, 'little')  # 2B number of elements to read
```
Same also for line 82 above